pkg/api/types.go

```diff
@@ -403,6 +404,13 @@ const (
 	PodUnknown PodPhase = "Unknown"
 )
 
+type PodCondition string
+
+// These are the valid service statuses of pods.
+const (
+	PodInService = "InService"
```

Comment:
Note that if we were to represent partial conditions, we'd need to ignore containers without probes defined rather than treat them as successful.

To be clear: I think we should put off reporting details about partial conditions, but am just thinking about likely future direction.

Another issue with respect to readiness is how to deal with multi-port services (#1802, #2585). If we remove pods that aren't ready from endpoints as in this PR, how would that affect multi-port services? I'm willing to punt on that for this PR. If the user specified a readiness probe using a named port on the container, we could map it back to a list of ports in the endpoint (so, per-port readiness, though only one port per container).